eminentdomain.game.php

```diff
@@ -1241,6 +1241,13 @@ function next_RoleDispatch($player_id) {
                     return; // cannot go pre-discard yet
                 }
             }
+
+            $leader = $this->getGameStateValue('leader');
```

Comment:
I will change to
``
            $leader = $this->getGameStateValue('leader');
            if ($leader == $player_id && $this->getUntrivialDiscardOrActivate($color) && !$discardPlayed && $rolePlayed >= $roleAllowerd ) {
                $this->gamestate->nextState('discard'); // pre-discard
                return;
            }